framework/include/postprocessors/Postprocessor.h

```diff
@@ -71,6 +71,8 @@ class Postprocessor : public OutputInterface,
    */
   const std::string & PPName() const { return _pp_name; }
 
+  virtual bool hasBlocks(SubdomainID /* id */) const override { return true; }
```

Comment:
Yep, otherwise you get an error like "this functor can't be block-restricted to whatever". @lindsayad can verify that is what should be done here.

Reply:
umm. Where do you get this?
We could wrap a PP functor in a PiecewiseByBlockFunctor to go around this, maybe

Reply:
This is the base class implementation of `hasBlocks` which he is trying to avoid
```c++
  virtual bool hasBlocks(SubdomainID /* id */) const
  {
    mooseError("Block restriction has not been implemented for functor " + functorName());
    return false;
  }
```

Reply:
but it s correct. We dont block restrict most PPs, only some. 
Which routine is calling this hasBlocks?

Reply:
A FaceArg evalution can easily trigger a `hasBlocks` call
```c++
template <typename T>
bool
FunctorBase<T>::hasFaceSide(const FaceInfo & fi, const bool fi_elem_side) const
{
  if (fi_elem_side)
    return hasBlocks(fi.elem().subdomain_id());
  else
    return fi.neighborPtr() && hasBlocks(fi.neighbor().subdomain_id());
}

template <typename T>
FaceArg
FunctorBase<T>::checkFace(const Moose::FaceArg & face) const
{
  const Elem * const elem = face.face_side;
  const FaceInfo * const fi = face.fi;
  mooseAssert(fi, "face info should be non-null");
  auto ret_face = face;
  bool check_elem_def = false;
  bool check_neighbor_def = false;
  if (!elem)
  {
    if (!hasFaceSide(*fi, true))
    {
      ret_face.face_side = fi->neighborPtr();
      check_neighbor_def = true;
    }
    else if (!hasFaceSide(*fi, false))
    {
      ret_face.face_side = fi->elemPtr();
      check_elem_def = true;
    }
  }
```

Reply:
alright seems we have no choice. I m ok with that